cmd/tsidp/tsidp_test.go

```diff
@@ -824,3 +826,139 @@ func TestExtraUserInfo(t *testing.T) {
 		})
 	}
 }
+
+func TestFunnelClientsPersistence(t *testing.T) {
+	testClients := map[string]*funnelClient{
+		"test-client-1": {
+			ID:          "test-client-1",
+			Secret:      "test-secret-1",
+			Name:        "Test Client 1",
+			RedirectURI: "https://example.com/callback",
+		},
+		"test-client-2": {
+			ID:          "test-client-2",
+			Secret:      "test-secret-2",
+			Name:        "Test Client 2",
+			RedirectURI: "https://example2.com/callback",
+		},
+	}
+
+	testData, err := json.Marshal(testClients)
+	if err != nil {
+		t.Fatalf("failed to marshal test data: %v", err)
+	}
+
+	tmpFile := t.TempDir() + "/oidc-funnel-clients.json"
+	if err := os.WriteFile(tmpFile, testData, 0600); err != nil {
+		t.Fatalf("failed to write test file: %v", err)
+	}
+
+	t.Run("step1_load_from_existing_file", func(t *testing.T) {
+		srv := &idpServer{}
+
+		// Simulate the funnel clients loading logic from main()
+		srv.funnelClients = make(map[string]*funnelClient)
```

Comment:
is this needed? The code looks like it's already doing `mak.Set(&s.funnelClients, clientID, &newClient)`

Reply:
oh, line 863. sorry, ☕ 

but I think that decode will still work anyway.